enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,495 @@
+mechanism and its promotion to beta eventually.
+
+### User Stories [optional]
+
+#### Story 1
+
+As an average customer, I want to have a default audit policy which does not cause
+noticeable resource overhead (cpu and disk).
+
+#### Story 2
+
+As any customer, I don't want to leak sensible information of resources like secrets,
+oauth tokens, etc. in audit logs.
+
+#### Story 3
+
+As a security and regulator demanding customer, I want to log **every** non-sensible resource
+change up to request level detail. 
+
+#### Story 4
+
+As a security and regulator demanding customer, I want to log every non-sensible resource
+change up to request level detail **from non-OpenShift components**.
+
+#### Story 5
+
+As a security and regulator demanding customer, I want to log every non-sensible resource
+change up to request level detail, **from clients outside of the cluster (non-service-accounts)**.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The proposed API looks like this: 
+
+```yaml
+kind: APIServer
+apiVersion: config.openshift.io/v1
+spec:
+  ...
+  audit:
+    profile: Default | OauthUserRequests | NonOpenshiftRequests | AllRequests
+```
+
+In the future, we could add more, even more detailed profiles of logging responses, if customers need this:
+
+```yaml
+    profile: ... | OauthUserResponses | NonOpenShiftResponses | AllResponses
+```
+
+But we are not going to add these without a strong business case.
+
+### Risks and Mitigations
+
+- going beyond the proposed profiles, e.g. by filtering of API groups, user agents or namespaces, yields the risk that the
+  defined semantics do not map to the future [upstream dynamic audit](https://github.com/kubernetes/enhancements/blob/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/sig-auth/0014-dynamic-audit-configuration.md)
+  API types. In the file-based API of kube-apiserver, which implements powerful, but controversial rule behaviour, this kind
+  filters are possible, but it also turned out as the biggest blocker for moving it toward the dynamic variant as part of 
+  the Kubernetes REST API. We can not and must not follow this failure as it is big risk to the maintainability and future 
+  feasibility of this OpenShift API.
+
+## Design Details
+
+The profile defined in the `APIServer` singleton instance named `cluster` will be translated into the respective file-based audit policy of
+
+- kube-apiserver
+- openshift-apiserver
+- oauth-apiserver
+
+by their respective operators.
+
+The profile translate like this:
+
+- `Default` translates to the policy given above.
+- `OauthUserRequests` translates to the following tail (with the head the same as the default policy):
+
+  ```yaml
+      ... # common head
+
+      # exclude resources where the body is security-sensitive
+      - level: MetaData
+        resources:
```

Comment:
oauthclients should appear here and later as well

Reply:
fixed